src/main/java/org/folio/services/transactions/TransactionTotalService.java

```diff
@@ -0,0 +1,82 @@
+import org.folio.rest.core.models.RequestContext;
+import org.folio.rest.core.models.RequestEntry;
+import org.folio.rest.jaxrs.model.TransactionTotal;
+import org.folio.rest.jaxrs.model.TransactionTotalCollection;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.function.Function;
+
+import static one.util.streamex.StreamEx.ofSubLists;
+import static org.folio.rest.util.HelperUtils.collectResultsOnSuccess;
+import static org.folio.rest.util.HelperUtils.convertIdsToCqlQuery;
+import static org.folio.rest.util.ResourcePathResolver.TRANSACTION_TOTALS;
+import static org.folio.rest.util.ResourcePathResolver.resourcesPath;
+
+@Log4j2
+public class TransactionTotalService {
+
+  private static final int MAX_FUND_PER_QUERY = 5;
+
+  private final RestClient restClient;
+
+  public TransactionTotalService(RestClient restClient) {
+    this.restClient = restClient;
+  }
+
+  public Future<List<TransactionTotal>> getTransactionsFromFunds(List<String> fundIds, String fiscalYearId,
+                                                                 List<TransactionTotal.TransactionType> trTypes, RequestContext requestContext) {
+    return getTransactionsFromOrToFunds(fundIds, fiscalYearId, trTypes, "from", requestContext);
+  }
+
+  public Future<List<TransactionTotal>> getTransactionsToFunds(List<String> fundIds, String fiscalYearId,
+                                                               List<TransactionTotal.TransactionType> trTypes, RequestContext requestContext) {
+    return getTransactionsFromOrToFunds(fundIds, fiscalYearId, trTypes, "to", requestContext);
+  }
+
+  private Future<List<TransactionTotal>> getTransactionsFromOrToFunds(List<String> fundIds, String fiscalYearId,
+                                                                      List<TransactionTotal.TransactionType> trTypes, String direction, RequestContext requestContext) {
+    return collectResultsOnSuccess(ofSubLists(new ArrayList<>(fundIds), MAX_FUND_PER_QUERY)
+      .map(ids -> getTransactionsByFundChunk(ids, fiscalYearId, trTypes, direction, requestContext)
+        .map(transactions -> filterFundIdsByAllocationDirection(fundIds, transactions, direction)))
+      .toList())
+      .map(lists -> lists.stream().flatMap(Collection::stream).toList());
+  }
+
+  private Future<List<TransactionTotal>> getTransactionsByFundChunk(List<String> fundIds, String fiscalYearId,
+                                                                    List<TransactionTotal.TransactionType> trTypes, String direction, RequestContext requestContext) {
+    String fundIdField = "from".equals(direction) ? "fromFundId" : "toFundId";
+    String fundQuery = convertIdsToCqlQuery(fundIds, fundIdField, "==", " OR ");
+    List<String> trTypeValues = trTypes.stream().map(TransactionTotal.TransactionType::value).toList();
+    String trTypeQuery = convertIdsToCqlQuery(trTypeValues, "transactionType", "==", " OR ");
```

Comment:
from method name we are incorrectly using this by setting type value, better to use other method or create new one which take value and fieldName only